cmd/snap-confine/mount-support.c

```diff
@@ -404,6 +404,49 @@ static void sc_bootstrap_mount_namespace(const struct sc_mount_config *config)
 			    dst);
 		}
 	}
+	// Since we mountec /etc from the host filesystem the /etc/os-release file
+	// (which may not be present) may be a symbolic link. It is expected that
+	// we see the real /etc/os-release file so let's do our best to show
+	// /etc/os-release as it looks like on the clasic distribution.
+	//
+	// If the symbolic link in /etc is pointing to /usr/lib/os-release then
+	// bind mount /usr/lib/os-release over itself in the core snap. This way
+	// the symlink (which we cannot over-bind mount anything) will work as
+	// expected.
+	//
+	// https://bugs.launchpad.net/canonical-livepatch-client/+bug/1667470
+	if (config->on_classic) {
```

Comment:
`/etc/os-release` it is standard enough now (not just a systemd thing, it's a pretty good thing in general) and it is the preferred way to figure out what people are running on.